parking/backend/db/dbaccess.py

```diff
@@ -74,3 +81,26 @@ class DbAccess(object):
                 logger.warning("Tried to allocate user : '{}' when they already had an allocation.".format(user_id))
                 return False
         return True
+
+    async def get_parking_lot_allocations(self, park_id: int) -> List[Record]:
+        async with self.pool.acquire() as conn:
+            records: List[Record] = await conn.fetch(c.ALLOCATIONS_SELECT, park_id)
+        return records
+
+    async def delete_allocation(self, user_id: str) -> None:
+        async with self.pool.acquire() as conn:
+            async with conn.transaction():
+                park_id: int = await conn.fetchval(c.ALLOCATIONS_DELETE, user_id)
+                if park_id:
+                    await conn.fetchval(c.PARKINGLOTS_DECREMENT_ALLOCATION, park_id)
```

Comment:
Big issue here as you're doing your delete and decrement in separate database transactions which can result in an inconsistent database. You should set up an explicit transaction that does both queries at once using `BEGIN`/`COMMIT`.

Reply:
The context manager (```async with conn.transaction()```) creates the transaction that the operations are performed in.

I realised that I can use ```await conn.execute(c.PARKINGLOTS_DECREMENT_ALLOCATION, park_id)``` instead of fetchval so have switched it